pycaret/containers/models/time_series.py

```diff
@@ -2243,18 +2225,17 @@ class XGBCdsDtContainer(CdsDtContainer):
     model_type = TSModelTypes.TREE
 
     def return_regressor_class(self):
-        try:
+        if _check_soft_dependencies("xgboost", extra="models", severity="warning"):
             import xgboost
-        except ImportError:
-            self.logger.warning("Couldn't import xgboost.XGBRegressor")
-            self.active = False
-            return
 
-        xgboost_version = tuple([int(x) for x in xgboost.__version__.split(".")])
-        if xgboost_version < (1, 1, 0):
-            self.logger.warning(
-                f"Wrong xgboost version. Expected xgboost>=1.1.0, got xgboost=={xgboost_version}"
-            )
+            xgboost_version = tuple([int(x) for x in xgboost.__version__.split(".")])
+            if xgboost_version < (1, 1, 0):
```

Comment:
We could use `version.parse` while we are at it (may need an import):
```suggestion
            xgboost_version = version.parse(xgboost.__version__)
            if xgboost_version < version.parse("1.1.0"):
```

Reply:
Change made